pkg/xray/watches.go

```diff
@@ -0,0 +1,246 @@
+package xray
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"log"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type WatchGeneralData struct {
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+	Active      bool   `json:"active,omitempty"`
+}
+
+type WatchFilterValue struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type WatchFilter struct {
+	Type  string `json:"type,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type WatchProjectResource struct {
+	Type            string        `json:"type,omitempty"`
+	BinaryManagerId string        `json:"bin_mgr_id,omitempty"`
+	Filters         []WatchFilter `json:"filters,omitempty"`
+	// Watch a repo
+	Name string `json:"name,omitempty"`
+}
+
+type WatchProjectResources struct {
+	Resources []WatchProjectResource `json:"resources,omitempty"`
+}
+
+type WatchAssignedPolicy struct {
+	Name string `json:"name,omitempty"`
+	Type string `json:"type,omitempty"`
+}
+
+type Watch struct {
+	GeneralData      WatchGeneralData      `json:"general_data,omitempty"`
+	ProjectResources WatchProjectResources `json:"project_resources,omitempty"`
+	AssignedPolicies []WatchAssignedPolicy `json:"assigned_policies,omitempty"`
+}
+
+func unpackWatch(d *schema.ResourceData) *Watch {
+	watch := new(Watch)
+
+	gd := WatchGeneralData{
+		Name: d.Get("name").(string),
+	}
+	if v, ok := d.GetOk("description"); ok {
```

Comment:
I believe if we remove `omitempty` then we won't need to do all these checking before unpacking these fields. @chb0github ?

Reply:
correct